File: Frontend/apps/astrogame_app/lib/models/technologies/stellar_object_dependent_technology_upgrade.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/technologies/technology_upgrade.dart';

import 'package:json_annotation/json_annotation.dart';

part 'stellar_object_dependent_technology_upgrade.g.dart';

@GuidConverter()
@JsonSerializable()
class StellarObjectDependentTechnologyUpgrade extends TechnologyUpgrade {
  late Guid stellarObjectId;

  StellarObjectDependentTechnologyUpgrade();

  factory StellarObjectDependentTechnologyUpgrade.fromJson(
          Map<String, dynamic> json) =>
      _$StellarObjectDependentTechnologyUpgradeFromJson(json);
  Map<String, dynamic> toJson() =>
      _$StellarObjectDependentTechnologyUpgradeToJson(this);
}

File: Backend/AstroGame.Shared/Models/Technologies/StellarObjectDependentTechnologyUpgrade.cs
using System;
using AstroGame.Shared.Models.Stellar.BaseTypes;
using Newtonsoft.Json;

namespace AstroGame.Shared.Models.Technologies
{
    public class StellarObjectDependentTechnologyUpgrade : TechnologyUpgrade
    {
        public Guid StellarObjectId { get; set; }

        [JsonIgnore] public virtual StellarObject StellarObject { get; set; }
    }
}